sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ordering.scala

```diff
@@ -57,20 +57,10 @@ class InterpretedOrdering(ordering: Seq[SortOrder]) extends BaseOrdering {
         return if (order.nullOrdering == NullsFirst) 1 else -1
       } else {
         val comparison = order.dataType match {
-          case dt: AtomicType if order.direction == Ascending =>
-            dt.ordering.asInstanceOf[Ordering[Any]].compare(left, right)
-          case dt: AtomicType if order.direction == Descending =>
-            - dt.ordering.asInstanceOf[Ordering[Any]].compare(left, right)
-          case a: ArrayType if order.direction == Ascending =>
-            a.interpretedOrdering.asInstanceOf[Ordering[Any]].compare(left, right)
-          case a: ArrayType if order.direction == Descending =>
-            - a.interpretedOrdering.asInstanceOf[Ordering[Any]].compare(left, right)
-          case s: StructType if order.direction == Ascending =>
-            s.interpretedOrdering.asInstanceOf[Ordering[Any]].compare(left, right)
-          case s: StructType if order.direction == Descending =>
-            - s.interpretedOrdering.asInstanceOf[Ordering[Any]].compare(left, right)
-          case other =>
-            throw QueryExecutionErrors.orderedOperationUnsupportedByDataTypeError(other)
+          case dt if order.direction == Ascending =>
+            PhysicalDataType.ordering(dt).compare(left, right)
```

Comment:
We really shouldn't resolve the ordering during execution. I know the old code does this, but the old is a bit quicker than the new code. It would be better to resolve it once as a field and then keep reusing it.

Reply:
Right. Move this to a lazy val to initialize it once.